src/libos/src/process/syscalls.rs

```diff
@@ -142,20 +144,43 @@ pub fn do_futex(
         Ok(val as usize)
     };
 
-    let get_futex_timeout = |timeout| -> Result<Option<FutexTimeout>> {
-        let timeout = timeout as *const timespec_t;
-        if timeout.is_null() {
-            return Ok(None);
-        }
-        let ts = timespec_t::from_raw_ptr(timeout)?;
-        ts.validate()?;
-        // TODO: use a secure clock to transfer the real time to monotonic time
-        let clock_id = if futex_flags.contains(FutexFlags::FUTEX_CLOCK_REALTIME) {
-            ClockID::CLOCK_REALTIME
+    let get_futex_timeout = |timeout| -> Result<Option<Duration>> {
+        // Sanity checks
+        let mut timeout: Duration = {
+            let timeout = timeout as *const timespec_t;
+            if timeout.is_null() {
+                return Ok(None);
+            }
+            let ts = timespec_t::from_raw_ptr(timeout)?;
+            ts.validate()?;
+            ts.as_duration()
+        };
+
+        // Only FUTEX_WAIT takes the timeout input as relative
+        let is_absolute = if futex_op != FutexOp::FUTEX_WAIT {
+            true
         } else {
-            ClockID::CLOCK_MONOTONIC
+            false
         };
-        Ok(Some(FutexTimeout::new(clock_id, ts)))
+        if is_absolute {
+            // TODO: use a secure clock to transfer the real time to monotonic time
+            let clock_id = if futex_flags.contains(FutexFlags::FUTEX_CLOCK_REALTIME) {
```

Comment:
Actually, I have implemented a commit using just one timeout, see [here](https://github.com/liqinggd/occlum/commit/451f2047774f5b665204b61ebe8ce87f8edfbc28) for the detail.
The reason why I choose to add the `clockbit` argument is that currently, we do not have a secure clock or vdso, so the `clock_get_time` Ocall is too slow for `futex`, so I add the TODO here.